packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -245,15 +250,22 @@ class SnackBar extends StatelessWidget {
     final MediaQueryData mediaQueryData = MediaQuery.of(context);
     assert(animation != null);
     final ThemeData theme = Theme.of(context);
+    final ColorScheme colorScheme = theme.colorScheme;
     final SnackBarThemeData snackBarTheme = theme.snackBarTheme;
-    // TODO(rami-a): Use a light theme if the app has a dark theme, https://github.com/flutter/flutter/issues/31418
-    final ThemeData darkTheme = ThemeData(
-      brightness: Brightness.dark,
-      accentColor: theme.accentColor,
-      accentColorBrightness: theme.accentColorBrightness,
+    final bool isThemeDark = theme.brightness == Brightness.dark;
+
+    // SnackBar uses a theme that is the opposite brightness from
+    // the surrounding theme.
+    final ThemeData inverseTheme = ThemeData(
+      brightness: isThemeDark ? Brightness.light : Brightness.dark,
+      backgroundColor: isThemeDark
+        ? colorScheme.onSurface
+        : Color.alphaBlend(colorScheme.onSurface.withOpacity(0.80), colorScheme.surface),
+      accentColor: isThemeDark ? colorScheme.primaryVariant : colorScheme.secondary,
```

Comment:
Shouldn't this inverseTheme's colorScheme have a similar secondary color? Shouldn't it have a new color scheme?

Reply:
Maybe open an issue about the need for a proper inverse color scheme here.